core/modules/user.py

```diff
@@ -946,6 +992,33 @@ def getAuthMethods(self, *args, **kwargs):
 
         return json.dumps(res)
 
+    @exposed
+    def trigger(self, action: str, key: str, skey: str):
+        current.request.get().response.headers["Content-Type"] = "application/json"
+
+        # only "root" users are allowed to trigger maintenance functions!
+        if not ((cuser := current.user.get()) and "root" in cuser["access"]):
```

Comment:
This should be configurable in the adminInfo.

Reply:
But how?

Reply:
I'm not sure about the meaning of access within custom access. Does one of the access roles have to be satisfied or all of them?


But something like this should do it?
```suggestion
        access = self.adminInfo[action]["access"]
        if not ((cuser := current.user.get()) and any(role in cuser["access"] for in access):
```